### PR TITLE
scripted diff: fix E275 lint

### PR DESCRIPTION
**Pull request by willcl-ark**
Fix E275 errors in lint-python.py with flake8 5.0.4

After updating my venv to Python 3.6.15 and reinstalling flake8 (at the current version, 5.0.4), it now throws quite a few [E275](https://www.flake8rules.com/rules/E275.html) which are fixed with this scripted diff.

### CODE REVIEW

**Comment by fanquake** [comment]
Duplicate of #26257.

**Comment by MarcoFalke** [comment]
Not sure. To me it seems odd to enforce this whitespace but no other. If you want to enforce whitespace, it might be best to discuss for all whitespace, unless there is a reason to do it for only this one.

**Comment by willcl-ark** [comment]
First, sorry I missed duplicating #26257 (but I also think a scripted diff is the best way to make this change).

@MarcoFalke which other whitespace did you have in mind?

IMO if we are using a specific linter then we should try to ensure those lints pass (or are disabled) and not more without adding additional lints for it. Otherwise if we change "all" whitespace today, we are likely to miss more of the same un-linted whitespace errors being introduced in the future.

**Comment by MarcoFalke** [comment]
> scripted diff is the best way

The scripted diff as of now may modify out-of-git-tree files with the wildcard. You'd have to use `git` to get the file list. Also, it doesn't pass CI as of now.

> @MarcoFalke which other whitespace did you have in mind?

Anything you could come up with. Pretty sure there are other whitespace/formatting checks in flake. If not, you can take one from `black` as example. (Not saying you should do this, just presenting this as a reason against).

> specific linter then we should try to ensure those lints pass

Yes, the only supported version is the one installed by CI. Obviously we can't support future versions that may not have been released yet or even written.

> errors being introduced in the future

Yes, so fixing this without enforcement seems doubly-bad. (Again, not saying you should do this, just presenting this as another reason against).

File: test/functional/rpc_signrawtransactionwithkey.py [hunk 1]
```diff
@@ -81,7 +81,7 @@ def witness_script_test(self):
         # Create a new P2SH-P2WSH 1-of-1 multisig address:
         eckey = ECKey()
         eckey.generate()
-        embedded_privkey = bytes_to_wif(eckey.get_bytes())
+        embedded_privkey = bytes_to_wif (eckey.get_bytes())
```
Comment (MarcoFalke) [inline test/functional/rpc_signrawtransactionwithkey.py:84]: Pretty sure this is wrong.

**Review by maflcko — COMMENTED** [review]
Taking a look at the number of conflict, see also my template answer:

Thank you for your contribution. While this stylistic change makes sense on its own, it comes at a cost and risk for the project as a whole. The weak motivation for the change does not justify the burden that it places on the project. A burden could be any of the following:

* Time spent on review
* Accidental introduction of bugs
* (Silent) merge conflicts, either in the branch or a backport branch. Those conflicts demand further developer and reviewer time or introduce bugs.

For more information about refactoring changes and stylistic cleanup, see
* https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#refactoring
* https://github.com/bitcoin/bitcoin/blob/master/.github/PULL_REQUEST_TEMPLATE.md
* https://github.com/bitcoin/bitcoin/issues/15465

Generally, if the style is not mentioned nor enforced by the developer notes, we leave it up to the original author to pick whatever fits them best personally and then leave it that way until the line is touched for other reasons.

Let me know if you have any questions.

**Comment by willcl-ark** [comment]
OK thanks.

Well, I'll switch this to use git files and makes sure it passes CI and leave it as an option vs #26257.

File: test/functional/feature_nulldummy.py [hunk 2]
```diff
@@ -72,7 +72,7 @@ def create_transaction(self, *, txid, input_details=None, addr, amount, privkey)
     def run_test(self):
         eckey = ECKey()
         eckey.generate()
-        self.privkey = bytes_to_wif(eckey.get_bytes())
+        self.privkey = bytes_to_wif (eckey.get_bytes())
```
Comment (flack) [inline test/functional/feature_nulldummy.py:75]: these seem to be matched by accident by the `if(` regex

**Comment by willcl-ark** [comment]
> Thank you for your contribution. While this stylistic change makes sense on its own, it comes at a cost and risk for the project as a whole. The weak motivation for the change does not justify the burden that it places on the project.

OK.  I will just apply the patch locally as I'm using newer flake8.

**Comment by MarcoFalke** [comment]
No objection, if you want to bump the CI version and add the rule to the excludes. Should be non-controversial.

**Comment by willcl-ark** [comment]
OK I can re-open with a bump to the CI in case the scripted diff is preferred over #26257

**Comment by jonatack** [comment]
This seems less pythonic and at first glance I'm not sure why it would be a better solution than the PR with two ACKs. I do think it's preferable to give review feedback than open competing PRs.

**Comment by MarcoFalke** [comment]
I said "add the rule to the excludes" should be uncontroversial, not the scripted-diff, which is still broken in many ways, as I pointed out already.

**Comment by willcl-ark** [comment]
Ah OK I understand. I will review the PR from @jonatack which I agree, without the brackets, is more pythonic.

**Comment by MarcoFalke** [comment]
E301, E302, E303, ... and others aren't enforced either, so why would E275 be important enough  to enforce?

**Comment by willcl-ark** [comment]
When python was bumped to 3.6.15 I made a new venv, installed flake8 et. al. (with a simple `pip install flake8` -- not using the pinned versions from ci/lint/04_install.sh) and ran the linter, which resulted in the E275 errors that this change fixed, but no E301, E302, E303, ... errors. So I suppose newer flake8 is checking E275 by default?

I'm not sure if we expect people to manually extract and locally install pinned versions from that ci script (it is linked from the readme), add a requirements.txt to `test/`, or leave the status quo. But I suspected I would not be the only person to encounter this (and had I used GitHub's search function like a good netizen I would have seen I was correct without needing to open this conflicting PR!)

**Pull request closed without merging**